modules/layers/src/column-layer/column-layer.js

```diff
@@ -91,52 +102,75 @@ export default class ColumnLayer extends Layer {
     super.updateState({props, oldProps, changeFlags});
     if (props.fp64 !== oldProps.fp64 || props.diskResolution !== oldProps.diskResolution) {
       const {gl} = this.context;
-      if (this.state.model) {
-        this.state.model.delete();
+      if (this.state.models) {
+        this.state.models.forEach(model => model.delete());
       }
-      this.setState({model: this._getModel(gl)});
+      this.setState(this._getModels(gl));
       this.getAttributeManager().invalidateAll();
     }
-
     if (props.vertices !== oldProps.vertices) {
       this._updateVertices(props.vertices);
     }
   }
 
-  getGeometry(diskResolution) {
-    return new CylinderGeometry({
+  getGeometry(diskResolution, outline = false) {
+    return new ColumnGeometry({
       radius: 1,
       topCap: false,
       bottomCap: true,
+      outline,
       height: 2,
       verticalAxis: 'z',
       nradial: diskResolution,
       nvertical: 1
     });
   }
 
-  _getModel(gl) {
-    return new Model(
-      gl,
-      Object.assign({}, this.getShaders(), {
-        id: this.props.id,
-        geometry: this.getGeometry(this.props.diskResolution),
-        isInstanced: true,
-        shaderCache: this.context.shaderCache
-      })
-    );
+  _getModels(gl) {
+    const {id, filled, extruded, diskResolution} = this.props;
+
+    let filledModel;
+    let strokeModel;
+
+    if (filled) {
+      filledModel = new Model(
+        gl,
+        Object.assign({}, this.getShaders(), {
+          id: `${id}-top`,
+          geometry: this.getGeometry(diskResolution, drawLines.filled),
```

Comment:
not necessary to declare an extra variable `drawLines`. use a string `filled` is sufficient and more readable.